osc-server/src/main/java/org/osc/core/broker/service/SGConformService.java

```diff
@@ -0,0 +1,186 @@
+/*******************************************************************************
+ * Copyright (c) Intel Corporation
+ * Copyright (c) 2017
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.core.broker.service;
+
+import javax.persistence.EntityManager;
+
+import org.osc.core.broker.job.Job;
+import org.osc.core.broker.job.Job.JobCompletionListener;
+import org.osc.core.broker.job.JobEngine;
+import org.osc.core.broker.job.JobQueuer;
+import org.osc.core.broker.job.JobQueuer.JobRequest;
+import org.osc.core.broker.job.TaskGraph;
+import org.osc.core.broker.job.lock.LockObjectReference;
+import org.osc.core.broker.model.entities.job.JobRecord;
+import org.osc.core.broker.model.entities.virtualization.SecurityGroup;
+import org.osc.core.broker.service.persistence.OSCEntityManager;
+import org.osc.core.broker.service.persistence.SecurityGroupEntityMgr;
+import org.osc.core.broker.service.tasks.conformance.UnlockObjectMetaTask;
+import org.osc.core.broker.service.tasks.conformance.openstack.securitygroup.SecurityGroupCheckMetaTask;
+import org.osc.core.broker.util.TransactionalBroadcastUtil;
+import org.osc.core.broker.util.db.DBConnectionManager;
+import org.osc.core.common.job.TaskGuard;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.transaction.control.ScopedWorkException;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(service = {SGConformService.class})
+public class SGConformService {
```

Comment:
generally all services in OSC are service dispatchers....we should come up with a different name for the these classes SecurityGroupJobManager/Factory/?

Reply:
DeploymentSpecConformJobFactory, DistributedApplianceConformJobFactory. Honestly, I would love to abbreviate do DAConformJobFactory, etc.

Reply:
we can put specific guidelines like below

https://stackoverflow.com/a/7366332/2904615

Lets not use abbreviations if possible atleast for terms internal to OSC